From ceb8090fcc811da4f89d4c38f221294441b91487 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Mon, 29 Nov 2021 19:21:17 +0200
Subject: [PATCH] [SW-63939] gaudi2: elbi: configure axprot_ovrd reg

Configured CORE_LBW_AXPROT_OVRD of PCIE_ELBI_RR_MSTR_IF
as secured. This is to block transactions via ELBI interface.
ELBI RRs(PCIE_ELBI_RR_MSTR_IF) also configured for privilege access.
So any transaction via ELBI will be considered as secured and expected
to be blocked by ELBI RRs.

Change-Id: I080a974b525e55550710f5ce04f07234fe74bc1b
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 86ea237..a345421 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -28,6 +28,7 @@
 #include <sft0_hbw_rtr_if0_mstr_if_rr_shrd_hbw_regs.h>
 #include <dcore0_rtr2_masks.h>
 #include <dcore0_rtr2_regs.h>
+#include <pcie_elbi_rr_mstr_if_core_lbw_regs.h>
 
 /*
  * instances_num : Number of instances of RR to program.
@@ -402,6 +403,7 @@
 }
 
 #define EN_ARC_SEC_PRIV				0x3
+#define AXPROT_OVRD_SEC				0x1
 
 /* protection bits configurations to open some selective registers */
 static u32 gaudi2_protbits_data0[] = { 0x400, 0xc103 };
@@ -425,6 +427,12 @@
 		 mmPSOC_ARC1_MSTR_IF_CORE_HBW_AXPROT_OVRD);
 	regmap_writel(EN_ARC_SEC_PRIV,
 		 mmPSOC_ARC1_MSTR_IF_CORE_LBW_AXPROT_OVRD);
+}
+
+static void preboot_secure_elbi(void)
+{
+	/* make transactions via ELBI as secured */
+	regmap_writel(AXPROT_OVRD_SEC, mmPCIE_ELBI_RR_MSTR_IF_CORE_LBW_AXPROT_OVRD);
 }
 
 static int init_prot_bits_block(u32 index, u8 major, u8 minor, u8 sub_minor)
@@ -858,6 +866,8 @@
 {
 	preboot_secure_arc();
 
+	preboot_secure_elbi();
+
 	/* Initialize only if security is enabled */
 	if (!preboot_security_is_enabled())
 		return;
